### PR TITLE
Upgrades Torchvision, Lightning, Lightning Utilities and Gym[atari] and fixes imports.

### LINKED ISSUE #962: Cannot import name 'LightningLoggerBase' from 'pytorch_lightning.loggers'
## 🐛 Bug

When trying to import anything from pl_bolts, I get the error: `cannot import name 'LightningLoggerBase' from 'pytorch_lightning.loggers'`. 

### To Reproduce
I'm currently using Keggle's pytorch_lightning version 1.9.0, and I saw that from version 1.7.0 it has been deprecated in favour of `pytorch_lightning.loggers.logger.Logger` but I don't want to downgrade my pl version: is pl_bolts being being updated to reach the more recent versions of pl?

Thanks in advance for your help! 🙏🏻 

Having a pl version >=1.7.0
```py
!pip install lightning-bolts
from pl_bolts.datasets import DummyDataset
```
raises: `cannot import name 'LightningLoggerBase' from 'pytorch_lightning.loggers'`

### PR DESCRIPTION
**Pull request by wilderrodrigues**
## What does this PR do?

Currently, Lightning Bolts depends on a very old version of Lightning and on Torchvision 0.10. That hinders the development if one wants to stay up-to-date with the latest releases.

I would like to propose we update the versions of the packages and release a 0.7 version of Bolt 

Fixes #962

## Before submitting

- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [X] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [X] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by wilderrodrigues** [comment]
@Borda @akihironitta could you have a look at this, please?

**Comment by Borda** [comment]
> 

will do later this week... :chipmunk:

**Comment by cdeepali** [comment]
@wilderrodrigues, I saw following error while executing the tests:

```
ImportError while loading conftest '<mypath>/pytorch-lightning-bolts/tests/conftest.py'.
tests/conftest.py:8: in &lt;module&gt;
    from pytorch_lightning.trainer.connectors.signal_connector import SignalConnector
E   ImportError: cannot import name 'SignalConnector' from 'pytorch_lightning.trainer.connectors.signal_connector' 
```
For this could you add an update in your PR to use 

`from pytorch_lightning.trainer.connectors.signal_connector import _SignalConnector` instead of `from pytorch_lightning.trainer.connectors.signal_connector import SignalConnector` in tests/conftest.py.

**Comment by wilderrodrigues** [comment]
> 

Thanks for the check and the code snippet, @cdeepali . Doing it now. Also, sorry for the delay, been loaded with day job work and couldn't get here before.

**Comment by zhutmost** [comment]
Hi. Will this pull request be merged in few days?
Our projects depend on DataModules in bolts, but it looks that the current nightly-edge bolts version is not compatible with Lightning 2.0.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [unReleased] - 2022-MM-DD
+## [unReleased] - 2023-03-06
```
Comment (Borda) [inline CHANGELOG.md:7]:
```suggestion
## [unReleased] - 2023-MM-DD
```

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-numpy <1.24.0 # todo: seeing some compatibility issues
-pytorch-lightning >=1.7.0, <1.9.0
-lightning-utilities >=0.3.0, !=0.4.0  # this is needed for PL 1.7
-torchvision >=0.10.0  # todo: move to topic related extras
+numpy<=1.24.0 # todo: seeing some compatibility issues
+pytorch-lightning>=1.9.4
```
Comment (Borda) [inline requirements.txt:2]:
```suggestion
pytorch-lightning>=1.9.0
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@wilderrodrigues, this is a very important contribution! :rabbit: Can we help you?

**Comment by wilderrodrigues** [comment]
> @wilderrodrigues, this is a very important contribution! 🐰 Can we help you?

Hi @Borda ,

Been loaded with work lately. 

I will rebase the branch today and update the PR. Also check what's going on with all those test failures.

**Comment by Borda** [comment]
> I will rebase the branch today and update the PR. Also check what's going on with all those test failures.

just check if it would be easier to make a new PR and addressing API updates for each dependency separately... :)

**Comment by Borda** [comment]
I'll be more than happy if you finish this upgrade, just as there are quite some collisions, let's start from the master and make each upgrade of a package in a separate PR :rabbit:

Just for the record:
- Torchmetrics shall be updated in #1016
- Gym will be partially updated in #918 but more work is needed to get it done with the latest
- PL has open PR #1031 but seems that more work is also needed and this could be several PRs also

**Pull request closed without merging**